### PR TITLE
Feature/health check unit tests

### PR DESCRIPTION
**Pull request by ChanduCES**

### CODE REVIEW

File: src/EmployeeApp/EmployeeApp.API/Constants/Endpoints.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace EmployeeApp.API.Constants
+{
+    public static class Endpoints
```
Comment (vaishali3012) [inline src/EmployeeApp/EmployeeApp.API/Constants/Endpoints.cs]: Suggestion : You can keep classame as ApiRoutes to keep consistency

**Review by vaishali3012 — CHANGES_REQUESTED** [review]

File: src/EmployeeApp/EmployeeApp.API/Constants/Endpoints.cs (hunk 1 above)
Comment (ChanduCES) [reply src/EmployeeApp/EmployeeApp.API/Constants/Endpoints.cs]: Changes done

File: src/EmployeeApp/EmployeeApp.IntegrationTest/SwaggerIntegrationTests.cs [hunk 2]
```diff
@@ -16,9 +18,14 @@ public async Task SwaggerApi_WhenCalled_ShouldReturnOKStatus()
...
             //Act
             var actual = await _httpClient.GetAsync(ApiRoutes.SwaggerEndpoint);
+            var result = await actual.Content.ReadAsStringAsync();
+
+            var json = JsonConvert.DeserializeObject<JToken>(result);
+            string title = json[AppConstants.InfoJToken][AppConstants.TitleJToken].ToString();
 
             //Assert
             actual.StatusCode.Should().Be(HttpStatusCode.OK);
+            title.Should().BeEquivalentTo(AppConstants.EmployeeAppAPI);
```
Comment (dkumarjha999) [inline src/EmployeeApp/EmployeeApp.IntegrationTest/SwaggerIntegrationTests.cs]: suggestion: use Should().Be() to compare two values.

File: src/EmployeeApp/EmployeeApp.API/Constants/ApiRoutes.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace EmployeeApp.API.Constants
+{
+    public static class ApiRoutes
+    {
+        public const string BaseUrl = "api";
+        public const string HealthCheck = "HealthCheck";
+        public const string SwaggerEndpoint = "/swagger/v1/swagger.json";
```
Comment (rohitvipin) [inline src/EmployeeApp/EmployeeApp.API/Constants/ApiRoutes.cs:7]: this is not used anywhere in the api, if its only used in test please move there

File: src/EmployeeApp/EmployeeApp.API/Constants/AppConstants.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿namespace EmployeeApp.API.Constants
+{
+    public static class AppConstants
```
Comment (rohitvipin) [inline src/EmployeeApp/EmployeeApp.API/Constants/AppConstants.cs:3]: these are not used in the app

File: src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="7.0.9" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.5.0" />
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (rohitvipin) [inline src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj:14]: Why?

File: src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="7.0.9" />
```
Comment (rohitvipin) [inline src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj:10]: Why?

File: src/EmployeeApp/EmployeeApp.API/Program.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+using EmployeeApp.API.Constants;
+
+public partial class Program 
+{
+    private static void Main(string[] args)
+    {
+        var builder = WebApplication.CreateBuilder(args);
+
+        builder.Services.AddHealthChecks();
+        builder.Services.AddMvcCore();
```
Comment (rohitvipin) [inline src/EmployeeApp/EmployeeApp.API/Program.cs:10]: Why?

File: src/EmployeeApp/EmployeeApp.IntegrationTest/SwaggerIntegrationTests.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        }
+
+        [Fact]
+        public async Task SwaggerApi_WhenCalled_ShouldReturnOKStatus()
+        {
+            //Act
+            var actual = await _httpClient.GetAsync(ApiRoutes.SwaggerEndpoint);
+            var result = await actual.Content.ReadAsStringAsync();
+
+            var json = JsonConvert.DeserializeObject<JToken>(result);
```
Comment (rohitvipin) [inline src/EmployeeApp/EmployeeApp.IntegrationTest/SwaggerIntegrationTests.cs:23]: what you deserialize is a json? then rather just use JObject

Comment (ChanduCES) [reply src/EmployeeApp/EmployeeApp.IntegrationTest/SwaggerIntegrationTests.cs:23]: Using JObject

File: src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj (hunk 5 above)
Comment (ChanduCES) [reply src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj:14]: This is for the swagger integration

File: src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj (hunk 6 above)
Comment (ChanduCES) [reply src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj:10]: This is for the swagger integration

Comment (rohitvipin) [reply src/EmployeeApp/EmployeeApp.API/EmployeeApp.API.csproj:10]: Not required. Please remove and see. 
please don’t add things unless it’s really required. 
https://learn.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-swashbuckle?view=aspnetcore-7.0&tabs=visual-studio

**Pull request merged**